content_sync/serializers.py

```diff
@@ -153,28 +151,107 @@ def deserialize(
+    content_uuids = [
+        menu_item["identifier"]
+        for menu_item in hugo_menu_data
+        if is_valid_uuid(menu_item["identifier"])
+    ]
+    return {
+        website_content.text_id: website_content
+        for website_content in WebsiteContent.objects.filter(text_id__in=content_uuids)
+    }
+
+
+def _transform_hugo_menu_data(
+    website_content: WebsiteContent, site_config: SiteConfig
+) -> Tuple[dict, dict]:
+    """
+    Adds 'url' property to internal links in menu data, and namespaces the menu data under "menu" in the
+    resulting dict (ref: https://gohugo.io/content-management/menus/#add-non-content-entries-to-a-menu)
+
+    Returns two dicts: the metadata with "menu" widget fields removed, and the transformed menu data namespaced
+        under "menu"
+    """
+    config_item = site_config.find_item_by_name(website_content.type)
+    menu_fields = {
+        field["name"] for field in config_item.fields if field.get("widget") == "menu"
+    }
+    result_menu = {}
+    for field_name in website_content.metadata.keys():
+        if field_name not in menu_fields:
+            continue
+        menu_data = website_content.metadata[field_name]
+        uuid_content_map = _get_uuid_content_map(menu_data)
+        result_menu_items = []
+        for menu_item in menu_data:
+            updated_menu_item = menu_item
+            # Add/update the 'url' value if this is an internal link
+            if menu_item["identifier"] in uuid_content_map:
+                menu_item_content = uuid_content_map[menu_item["identifier"]]
+                updated_menu_item["url"] = get_destination_filepath(
+                    menu_item_content, site_config
+                )
+            result_menu_items.append(updated_menu_item)
+        result_menu[field_name] = result_menu_items
+    return (
+        dict_without_keys(website_content.metadata, *menu_fields),
+        ({"menu": result_menu} if result_menu else {}),
+    )
+
+
+class HugoMenuYamlFileSerializer(BaseContentFileSerializer):
+    """
+    HACK: Hugo-specific logic for properly transforming data if the "menu" widget is used
+
+    Serializer/deserializer class for Hugo menu files
+    """
+
+    def serialize(self, website_content: WebsiteContent) -> str:
+        non_menu_metadata, menu_data = _transform_hugo_menu_data(
+            website_content, self.site_config
+        )
+        return yaml.dump(
+            self.serialize_contents(
+                metadata={**menu_data, **non_menu_metadata}, title=website_content.title
+            ),
+            Dumper=yaml.Dumper,
+        )
+
+    def deserialize(
+        self, website: Website, filepath: str, file_contents: str
+    ) -> WebsiteContent:
+        parsed_file_data = yaml.load(file_contents, Loader=yaml.Loader)
+        if "menu" in parsed_file_data:
+            parsed_file_data = {
+                **parsed_file_data["menu"],
+                **dict_without_keys(parsed_file_data, "menu"),
```

Comment:
Since `"menu"` appears a few times in the code here, might be better to add a constant like `CONTENT_MENU_FIELD = "menu"` and use that constant instead.